File: application/app/src/main/java/tech/linuxcat/saes4/types/Airport.kt
package tech.linuxcat.saes4.types

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
@kotlinx.serialization.Serializable
data class Airport(var id : String, var nom : String, var ville_nom : String, var vols : List<Vol>) : Parcelable

File: API/model/airport.mjs
'use strict'

import Vol from "./vol.mjs"

//class Airport
export default class Airport{
    id
    nom

    ville_nom
    vols
    
    constructor (obj){
        this.id = obj.id || ""
        this.nom = obj.nom || ""
        this.ville_nom = obj.ville_nom || ""

        if (obj.vols!=null){
            this.vols = obj.vols.map(vol => new Vol(vol))
        }else{
            this.vols = []
        }
    }
}